File: app/src/main/java/x/chestnut/code/snippet/ui/recyclerView/itemDecorationSection/GroupInfo.java
package x.chestnut.code.snippet.ui.recyclerView.itemDecorationSection;

/**
 * <pre>
 *     author: Chestnut
 *     blog  : https://juejin.im/user/676954892408824
 *     time  : 2019/3/30 15:37
 *     desc  :
 *     thanks To:
 *     dependent on:
 *     update log:
 * </pre>
 */
public class GroupInfo {
    //Header的Id
    public int groupId;
    //Header的标题
    public String groupTitle;
    //ItemView 在组内的位置
    public int positionInGroup;
    // 组的成员个数
    public int groupSize;

    public boolean isFirstViewInGroup () {
        return positionInGroup == 0;
    }

    public boolean isLastViewInGroup() {
        return positionInGroup == groupSize - 1 && positionInGroup >= 0;
    }
}

File: app/src/main/java/x/chestnut/code/snippet/ui/recyclerView/itemDecorationSection/GroupInfo.kt
package x.chestnut.code.snippet.ui.recyclerView.itemDecorationSection

/**
 * <pre>
 * author: Chestnut
 * blog  : https://juejin.im/user/676954892408824
 * time  : 2019/3/30 15:37
 * desc  :
 * thanks To:
 * dependent on:
 * update log:
</pre> *
 */
class GroupInfo {

    //Header的Id
    var groupId = 0

    //Header的标题
    var groupTitle: String = ""

    //ItemView 在组内的位置
    var positionInGroup = 0

    // 组的成员个数
    var groupSize = 0
    val isFirstViewInGroup: Boolean
        get() = positionInGroup == 0
    val isLastViewInGroup: Boolean
        get() = positionInGroup == groupSize - 1 && positionInGroup >= 0
}